### ISSUE TITLE
Use Wordpress functions in Commands

### ISSUE THREAD

**Issue by tipytechnique** [themosis/framework#646]
Hi !

Is there a way to use Wordpress functions like ```get_post_meta()``` on my custom console commands ?

Wordpress make a lot of stuff being his functions so i don't really want to use Eloquent to get data, i want to stick to the Wordpress functions.

For the moment i'm calling the WP REST API but my command can take a lot of times with a complex script.

Thanks :)

**Comment by jlambe** [comment]
Hi, WordPress is unfortunenately not loaded on the `console` application. Not sure how it will work if you try to include `wp-load.php` inside your command to gain access to WordPress. The console application is at least loading the wordpress constants defined in the `config/wordpress.php` file.

**Comment by tipytechnique** [comment]
Ok thank you :), i'll use the WP REST API so, if including ```wp-load.php``` is good or if i find another way i'll re-open this issue to share my code.

**Comment by tipytechnique** [comment]
Hi, back to this issue as i found a way to use Wp core functions on console commands.

What i did : 

- On `console` file, in the root folder, i added this line : `require web_path('cms/wp-load.php');` after bootstraping the application : 
```
require __DIR__.'/vendor/autoload.php';

$app = require_once __DIR__.'/bootstrap/app.php';

require web_path('cms/wp-load.php');
```

- Then, on `wp-config.php`, i needed to make a small update : 
```
$app = require_once __DIR__.'/../bootstrap/app.php';
if ($app === true) {
    $app = app();
}
```
In the console context, as the `bootstrap/app.php` file is already required, on `wp-config.php`, `$app` is a boolean (true). In this case, `$app = app();` make sure that `$app` is an instance of `Themosis\Core\Application`.

I made some tests on my existings commands, and this is working fine. The only issue is that on small commands calling `WP REST API` is faster than loading Wordpress functions, we are loosing like 1". But on large commands that call a lot of times Wordpress functions, loading Worpdress is faster.

On my project, a large command that call a lot of times the `getMeta` and the `updateMeta` functions, with the `WP REST API` takes 1'35" to finish, while with loading Wp functions it takes 11".

@jlambe what's your thought about this ?

**Comment by tipytechnique** [comment]
Hi, back to this issue as i found a way to use Wp core functions on console commands.

What i did : 

- On `console` file, in the root folder, i added this line : `require web_path('cms/wp-load.php');` after bootstraping the application : 
```
require __DIR__.'/vendor/autoload.php';

$app = require_once __DIR__.'/bootstrap/app.php';

require web_path('cms/wp-load.php');
```

- Then, on `wp-config.php`, i needed to make a small update : 
```
$app = require_once __DIR__.'/../bootstrap/app.php';
if ($app === true) {
    $app = app();
}
```
In the console context, as the `bootstrap/app.php` file is already required, on `wp-config.php`, `$app` is a boolean (true). In this case, `$app = app();` make sure that `$app` is an instance of `Themosis\Core\Application`.

I made some tests on my existings commands, and this is working fine. The only issue is that on small commands calling `WP REST API` is faster than loading Wordpress functions, we are loosing like 1". But on large commands that call a lot of times Wordpress functions, loading Worpdress is faster.

On my project, a large command that call a lot of times the `getMeta` and the `updateMeta` functions, with the `WP REST API` takes 1'35" to finish, while with loading Wp functions it takes 11".

@jlambe what's your thought about this ?

**Comment by jlambe** [comment]
@tipytechnique This is interesting. One thing though that I would like to make sure it is working is that the CLI tool should also work without having WordPress installed.

So the fact that requiring the WordPress `wp-load.php` file should not stop the process if WordPress hit an issue like database not yet installed for example.

But otherwise, this is definitely a good step 😃 👏

**Comment by tipytechnique** [comment]
Ok :) i'll open a draft PR soon with a first simple implementation of loading Wordpress on cli and investigate about keeping cli working without Wordpress installed.

Already thinking about enhancement, like let the user choose on their command to load Wordpress or not, no need to load all this stuff if the command is not using Wordpress functions.

**Comment by rumur** [comment]
Hi @tipytechnique !

Thanks for your PR, I believe it will help to many developers.
However, I'm sorry for my 5 cents, but what do you think about to rename the trait from 'LoadWordpress' -> `LoadWordPress` and the method, from `loadWordpress` to `loadWordPress`, it's a small thing, but it gives a consistency.

If you e.g. take a look at `\Themosis\Route\Middleware\WordPressBodyClass::class`.

@jlambe what do you think, does it make any sense?

**Comment by jlambe** [comment]
@tipytechnique I know it's a bit late but the `main` branch is now supporting WordPress for commands. In fact I went simpler by providing a `WordPressLoader` class that can be automatically injected in the command handle method. And instead of requiring the `wp-load.php` file, I used the the `wp-settings.php` file as we don't want to also load the HTTP kernel in a console context.

For information: @rumur 🙂
